awx/ui_next/src/screens/Project/shared/ProjectSubForms/SharedFields.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { Field } from 'formik';
+import CredentialLookup from '@components/Lookup/CredentialLookup';
+import FormField, { CheckboxField } from '@components/FormField';
+import { required } from '@util/validators';
+import FormRow from '@components/FormRow';
+import { FormGroup, Title } from '@patternfly/react-core';
+import styled from 'styled-components';
+
+export const SubFormTitle = styled(Title)`
+  --pf-c-title--m-md--FontWeight: 700;
+  grid-column: 1 / -1;
+`;
+
+export const UrlFormField = withI18n()(({ i18n, tooltip }) => (
+  <FormField
+    id="project-scm-url"
+    isRequired
+    label={i18n._(t`SCM URL`)}
+    name="scm_url"
+    tooltip={tooltip}
+    tooltipMaxWidth="350px"
+    type="text"
+    validate={required(null, i18n)}
+  />
+));
+
+export const BranchFormField = withI18n()(({ i18n, label }) => (
+  <FormField
+    id="project-scm-branch"
+    name="scm_branch"
+    type="text"
+    label={label}
+    tooltip={i18n._(t`Branch to checkout. In addition to branches,
+        you can input tags, commit hashes, and arbitrary refs. Some
+        commit hashes and refs may not be availble unless you also
+        provide a custom refspec.`)}
+  />
+));
+
+export const ScmCredentialFormField = withI18n()(
+  ({ i18n, setScmCredential, scmCredential }) => (
+    <Field
+      name="credential"
+      render={({ form }) => (
+        <CredentialLookup
+          credentialTypeId={scmCredential.typeId}
+          label={i18n._(t`SCM Credential`)}
+          value={scmCredential.value}
+          onChange={credential => {
+            form.setFieldValue('credential', credential.id);
+            setScmCredential({
+              ...scmCredential,
+              value: credential,
+            });
+          }}
+        />
+      )}
+    />
+  )
+);
+
+/*
+  PF Bug: FormGroup doesn't pass down className
+  Workaround is to wrap FormGroup with an extra div
+  Cleanup when upgraded to @patternfly/react-core@3.103.4
+*/
+export const ScmTypeOptions = withI18n()(
+  ({ i18n, scmUpdateOnLaunch, hideAllowOverride }) => (
+    <>
+      <div css="grid-column: 1/-1">
```

Comment:
I noticed that in a few places you're adding styles with the `css=` prop. There's also a few places in this pr where `style=` is used. When choosing between `css=`, `style=`, and wrappers made with `styled-components`, what criteria is being used?

Reply:
I prefer to use [`css=`](https://www.styled-components.com/docs/api#css-prop) because the prop is provided by styled-components. It allows you to write css the way you typically would in a stylesheet and supports everything that styled-components give you. The places where I used `styled=` was an overlook on my part.

Reply:
:+1: cool. I didn't realize that was part of our `styled-components` lib.